bot/exts/fun/off_topic_names.py

```diff
@@ -51,24 +60,129 @@ async def update_names(self) -> None:
         """Background updater task that performs the daily channel name update."""
         await self.bot.wait_until_guild_available()
 
-        try:
-            channel_0_name, channel_1_name, channel_2_name = await self.bot.api_client.get(
-                "bot/off-topic-channel-names", params={"random_items": 3}
-            )
-        except ResponseCodeError as e:
-            log.error(f"Failed to get new off-topic channel names: code {e.response.status}")
-            raise
-
-        channel_0, channel_1, channel_2 = (self.bot.get_channel(channel_id) for channel_id in CHANNELS)
-
-        await channel_0.edit(name=OTN_FORMATTER.format(number=0, name=channel_0_name))
-        await channel_1.edit(name=OTN_FORMATTER.format(number=1, name=channel_1_name))
-        await channel_2.edit(name=OTN_FORMATTER.format(number=2, name=channel_2_name))
+        channels = [await get_or_fetch_channel(channel_id) for channel_id in CHANNELS]
+        failed_renames = defaultdict(list)
+        successfully_renamed = set()
+        for attempt in range(MAX_RENAME_ATTEMPTS):
```

Comment:
I think the approach can be enhanced  a bit. 
We can fetch 3 names (number of attempts) for each channel.
We iterate over each channel, and try to rename it three times, then exit, that way we can increase the chance of a successful rename OP.

```py
number_of_names_to_fetch = MAX_ATTEMPS * len(CHANNELS)
channels = [get_or_fetch_channel(channel) for channel in CHANNELS]
names = iter(api_client.fetch_names(params={"random_items": number_of_names_to_fetch}))

renamed = set()
failed = set()

for channel in channels:
    attempt = 0
    while attempt < MAX_ATTEMPS:
        attempt += 1
        try:
            old_name = channel.name
            name = next(names)
            channel.update(name=name)
            renamed.add(channel.id)
            break
        except:
            # Catch your code here & deactivate the name
            # Reraise if not the right exception
            failed.add(f"- {name}\n") # Leverage the new markdown support to dsplay as a bullet list
            ...

# Simplify the handling of the failed renames.

if failed:
    unchanged = [channel.name for channel in channels if channel.id not in renamed]
    message = f"Couldn't rename the following off-topic channels: {', '.join(unchanged)}" \
              f"The following off-topic names were deactivated:\n{failed}", 
```

Reply:
Rewritten in 2f738ad1. Currently untested, but lmk what you think :)